python/ray/train/tests/test_mosaic_trainer.py

```diff
@@ -0,0 +1,146 @@
+from composer.loggers import InMemoryLogger
+from composer.algorithms import LabelSmoothing
+
+
+mean = (0.507, 0.487, 0.441)
+std = (0.267, 0.256, 0.276)
+cifar10_transforms = transforms.Compose(
+    [transforms.ToTensor(), transforms.Normalize(mean, std)]
+)
+# data_directory = "./data" ## TODO : remove the following line
+data_directory = "~/Desktop/workspace/data"
+train_dataset = datasets.CIFAR10(
+    data_directory, train=True, download=True, transform=cifar10_transforms
+)
+test_dataset = datasets.CIFAR10(
+    data_directory, train=False, download=True, transform=cifar10_transforms
+)
+
+scaling_config = ScalingConfig(num_workers=2, use_gpu=False)
+
+
+@pytest.fixture(autouse=True, scope="session")
+def ray_start_4_cpus():
+    address_info = ray.init(num_cpus=4)
+    yield address_info
+    # The code after the yield will run as teardown code.
+    ray.shutdown()
+
+
+def trainer_init_per_worker(**config):
+    BATCH_SIZE = 1024
+    # prepare the model for distributed training and wrap with ComposerClassifier for
+    # Composer Trainer compatibility
+    model = config.pop("model", torchvision.models.resnet18(num_classes=10))
+    model = ComposerClassifier(ray.train.torch.prepare_model(model))
+
+    # prepare train/test dataset
+    train_dataset = config.pop("train_dataset")
+    test_dataset = config.pop("test_dataset")
+
+    batch_size_per_worker = BATCH_SIZE // session.get_world_size()
+    train_dataloader = torch.utils.data.DataLoader(
+        train_dataset, batch_size=batch_size_per_worker, shuffle=True
+    )
+    test_dataloader = torch.utils.data.DataLoader(
+        test_dataset, batch_size=batch_size_per_worker, shuffle=True
+    )
+
+    train_dataloader = train.torch.prepare_data_loader(train_dataloader)
+    test_dataloader = train.torch.prepare_data_loader(test_dataloader)
+
+    evaluator = Evaluator(
+        dataloader=test_dataloader, label="my_evaluator", metrics=Accuracy()
+    )
+
+    # prepare optimizer
+    optimizer = composer.optim.DecoupledSGDW(
+        model.parameters(),
+        lr=0.05,
+        momentum=0.9,
+        weight_decay=2.0e-3,
+    )
+
+    if config.pop("eval", False):
+        config["eval_dataloader"] = evaluator
+
+    return composer.trainer.Trainer(
+        model=model, train_dataloader=train_dataloader, optimizers=optimizer, **config
+    )
+
+
+trainer_init_per_worker.__test__ = False
+
+
+def test_mosaic_e2e():
+    """Tests if the basic MosaicTrainer with minimum configuration runs and reports correct
+    Checkpoint dictionary.
+    """
+    trainer_init_config = {
+        "max_duration": "1ba",
+        "train_dataset": train_dataset,
+        "test_dataset": test_dataset,
+        "loggers": [InMemoryLogger()],
+        "algorithms": [LabelSmoothing()],
+    }
+
+    trainer = MosaicTrainer(
+        trainer_init_per_worker=trainer_init_per_worker,
+        trainer_init_config=trainer_init_config,
+        scaling_config=scaling_config,
+    )
+
+    trainer.fit()
```

Comment:
how long this test runs e2e ? If it's too long we can shrink our input dataset size

Reply:
For now, I only run for 1 batch, so the test time itself is not too long, but considering future tests, it might be helpful to shrink the input dataset size.

Reply:
Updated the test script dataset so that we shrink the dataset and run the trainer for 1 epoch instead of 1 batch; in this way any batch or epoch level callbacks would be called if needed.